test/tests/builder_tests.py

```diff
@@ -109,6 +110,36 @@ def test_setup_build_dir(self):
             self._cmp_files(test_archives/file,
                             test.builder.path/file)
 
+    def test_create_file(self):
```

Comment:
You need to test that you can't create a file outside of the build directory, and make sure you don't overrwrite a directory with a file.